src/librustc_middle/mir/mod.rs

```diff
@@ -1804,7 +1804,7 @@ pub enum ProjectionElem<V, T> {
     /// ```
     ConstantIndex {
         /// index or -index (in Python terms), depending on from_end
-        offset: u32,
+        offset: u64,
```

Comment:
I think you can now also remove a bunch of conversions around here:

https://github.com/rust-lang/rust/blob/bd0bacc694d7d8175804bb6f690cb846bfa4a9ee/src/librustc_mir/interpret/place.rs#L541